### PR TITLE
Add RabbitMQ as source and sink recipe

### PR DESCRIPTION
**Pull request by sabbyanandan**
Here's an attempt to demonstrate RabbitMQ is used as the source and sink along with RabbitMQ also used as the binder implementation. The goal of this recipe is to highlight how to configure two different RabbitMQ clusters for the same application.

- Demonstrate `rabbit | transform | rabbit` equivalent, but as standalone apps
- Highlight and speak to the configuration properties for the apps and as well the `|` (aka: binder)
- Attach RabbitMQ screenshots where appropriate

### CODE REVIEW

**Comment by sabbyanandan** [comment]
Thanks, @artembilan. I wanted to make sure that I'm not using extra unnecessary properties to connect to external Rabbit cluster. I had this worry of maybe using more than what's required.

File: content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+---
+path: 'recipes/rabbitmq/rabbit-source-sink/'
+title: 'RabbitMQ as Source and Sink'
+description: 'RabbitMQ as Source and Sink + RabbitMQ binder'
+---
+
+# RabbitMQ as Source and Sink
+
+Reading from and writing to RabbitMQ is a very common use-case. Primarily when the source and sink applications use spring Cloud Stream's RabbitMQ binder implementation, the configurations can be confusing. The goal of this recipe is to unpack the complexity step by step.
```
[the remark this answers is not in the document]
Comment (sabbyanandan) [reply content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md]: I think we aim for it here also. I will fix it.

File: content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+3. Setup external RabbitMQ cluster and prepare the cluster connection credentials.
+
+## Deployment
+
+With all the prerequisites from the previous step complete, we can now start the three applications.
+
+### Source
+
+```bash
+java -jar rabbit-source-rabbit-2.1.0.RELEASE.jar --server.port=9001 --rabbit.queues=sabbyfooz --spring.rabbitmq.addresses=amqp://<USER>:<PASSWORD>@<HOST> --spring.rabbitmq.port=5672 --spring.rabbitmq.username=<USER> --spring.rabbitmq.password=<PASSWORD> --spring.cloud.stream.binders.rabbitBinder.type=rabbit --spring.cloud.stream.binders.rabbitBinder.environment.spring.rabbitmq.addresses=amqp://guest:guest@127.0.0.1 --spring.cloud.stream.binders.rabbitBinder.environment.spring.rabbitmq.port=5672 --spring.cloud.stream.bindings.output.destination=rabzysrc
```
Comment (artembilan) [inline content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md]: Yeah, you indeed don't need `port` property if you use `addresses`. You just can join the host and port in a single property.

Comment (sabbyanandan) [reply content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md]: Thanks, @artembilan.

**Comment by sabbyanandan** [comment]
@ilayaperumalg: If you're also cool with it, could we merge it?

File: content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+3. Setup external RabbitMQ cluster and prepare the cluster connection credentials.
+
+## Deployment
+
+With all the prerequisites from the previous step complete, we can now start the three applications.
+
+### Source
+
+```bash
+java -jar rabbit-source-rabbit-2.1.0.RELEASE.jar --server.port=9001 --rabbit.queues=sabbyfooz --spring.rabbitmq.addresses=amqp://<USER>:<PASSWORD>@<HOST>:<PORT> --spring.rabbitmq.username=<USER> --spring.rabbitmq.password=<PASSWORD> --spring.cloud.stream.binders.rabbitBinder.type=rabbit --spring.cloud.stream.binders.rabbitBinder.environment.spring.rabbitmq.addresses=amqp://guest:guest@127.0.0.1:5672 --spring.cloud.stream.bindings.output.destination=rabzysrc
```
Comment (ilayaperumalg) [inline content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md]: Should we mention the properties that have prefix `spring.cloud.stream.binders` are the binder configuration properties and the `rabbitBinder` is just the configuration name?

File: content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+[[note]]
+| External RabbitMQ cluster credentials are supplied via `--spring.rabbitmq.*` properties.
+| The binder configurations are supplied via `--spring.cloud.stream.binders.rabbitBinder.environment.spring.rabbitmq.*` properties.
```
Comment (ilayaperumalg) [inline content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md]: The previous comment's explanation of the `rabbitBinder` being just the configuration name would probably fit here better.

**Comment by ilayaperumalg** [comment]
LGTM. Fixing the comments when merging this PR.

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as e22223b

**Comment by sabbyanandan** [comment]
Thanks, @ilayaperumalg!

**Pull request closed without merging**